contents/performance/WebPerformance.md

```diff
@@ -0,0 +1,79 @@
+
+In order to deliver a positive user experience, web developers must ensure that their applications meet acceptable performance standards.
+
+## Key Ideas in Improving Web Performance
+There are many factors which affect web performance. Here, we give an overview of some of the key ideas that can be used to improve the performance of your web application.
+
+**Reduce Javascript Payloads**
+
+Loading and executing Javascript is often [the slowest part](https://medium.com/@addyosmani/the-cost-of-javascript-in-2018-7d8950fbb5d4) of the page load process. Reducing the amount of Javascript that needs to be loaded thus significantly reduces the time taken for your site to load.
+
+Here are some steps that web developers can take to reduce the amount of Javascript that clients need to load:
+- Use [code splitting](https://developers.google.com/web/fundamentals/performance/optimizing-javascript/code-splitting/) to load only the Javascript required for the page being accessed.
+- Remove unused code (often parts of libraries that are not used in the application) with [tree shaking](https://developers.google.com/web/fundamentals/performance/optimizing-javascript/tree-shaking/).
+- Ensure that the delivered Javascript is [minified](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/javascript-startup-optimization/).
+- Create [performance budgets](https://infrequently.org/2017/10/can-you-afford-it-real-world-web-performance-budgets/) that specify the maximum size of Javascript payloads, and [enforce them](https://web.dev/fast/incorporate-performance-budgets-into-your-build-tools) to ensure that the size of your Javascript payloads remain under control.
+
+**Optimize Images**
+
+Images form a [significant portion](https://httparchive.org/reports/page-weight) of the resources loaded on web applications. To improve performance and speed up the site for users, web developers should thus try to reduce the amount of bandwidth used to load images.
+
+Images can be optimized for the web in the following ways:
+- Use the [correct file format](https://medium.com/beginners-guide-to-mobile-web-development/web-image-formats-googles-webp-17e2fe5fc53e) for your images.
+- [Compress your images](https://www.html5rocks.com/en/tutorials/speed/img-compression/) to reduce the amount of data that needs to be loaded. Be careful to strike a balance between image size and quality for optimal user experience.
+- Deliver [appropriately sized images](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images) based on the resolution of the client device. This prevents smaller (often mobile or lower-end) devices from wasting bandwidth loading large images that cannot be displayed at their full resolution on the device.
+
+**Progressive Loading**
```

Comment:
Can consider consistent phrasing to match sibling headings. e.g., `Use Progressive Loading`